### PR TITLE
Image mirror

### PR DESCRIPTION
**Pull request by Gl4di4torRr**
Shared Library for image mirror.

I changed the cross cluster promote to skopeo. If consumers are pointing directly to the master branch, this will break their builds. I hope no consumers are doing that.

### CODE REVIEW

**Comment by Gl4di4torRr** [comment]
@pabrahamsson @etsauer i have tested this shared lib... there are a couple of strange aspects to it (at least i think so) will you please review and let me know what you think?

**Comment by pabrahamsson** [comment]
@etsauer or @syvanen are either of you able to take a look at this PR?

File: vars/imageMirror.txt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```
+stage ("Image Mirror") {
+  steps {
+    imageMirror(sourceSecret: "namespace-sourcesecret",
+                sourceRegistry: "https://source-registry.com",
+                sourceAPI: "source-api.com",
+                destinationSecret: "namespace-destinationsecret",
+                destinationRegistry: "https://destination-registry.com",
+                destinationAPI: "destination-api.com",
```
Comment (syvanen) [inline vars/imageMirror.txt]: I tested this and it works but would Registry and API addresses ever differ?
Maybe hide the different url format into the code?

Comment (Gl4di4torRr) [reply vars/imageMirror.txt]: The only difference for me is `https://` but then we would be assuming the registry is https. I just wasnt sure that was a good approach. We could look for the protocol and not override it if it's there...

Comment (Gl4di4torRr) [reply vars/imageMirror.txt]: Take a look at my changes... I think that may work well...

File: vars/imageMirror.txt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+apiVersion: v1
+kind: Template
+labels:
+  template: ${SECRET_NAME}
+metadata:
+  annotations:
+    description: Cluster Credential Secret
+    tags: secret
+    version: 1.0.0
+  name: ${SECRET_NAME}
```
[the remark this answers is not in the document]
Comment (Gl4di4torRr) [reply vars/imageMirror.txt]: gave the template a hard coded name

File: vars/imageMirror.txt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+```
+
+Example Secret Template:
+
+```
+apiVersion: v1
+kind: Template
+labels:
+  template: ${SECRET_NAME}
```
[the remark this answers is not in the document]
Comment (Gl4di4torRr) [reply vars/imageMirror.txt]: harded coded the template value

**Comment by Gl4di4torRr** [comment]
@syvanen post holiday bump :)

**Comment by syvanen** [comment]
@Gl4di4torRr Finally had time to return to testing this PR. Just copied an image from test cluster to quay.io. So LGTM.

**Comment by Gl4di4torRr** [comment]
@syvanen thank you!

**Pull request merged**